lib/metadata_json_deps.rb

```diff
@@ -9,7 +9,11 @@ def initialize(cache = {})
 
     def get_module(name)
       name = PuppetForge::V3.normalize_name(name)
-      @cache[name] ||= PuppetForge::Module.find(name)
+      begin
+        @cache[name] ||= PuppetForge::Module.find(name)
+      rescue Faraday::ResourceNotFound
+        raise Exception.new("Dependency #{name} not found on forge.puppet.com")
```

Comment:
I'd prefer a specific exception for this. IMHO PuppetForge should do that. It already catches `Faraday::ResourceNotFound` many times and there is even a `PuppetForge::ModuleNotFound` class. It just isn't used. So I opened https://github.com/puppetlabs/forge-ruby/pull/104.

How about a compatible change:
```suggestion
        raise PuppetForge::ModuleNotFound.new("Dependency #{name} not found on forge.puppet.com")
```